contracts/MarketRegistry.sol

```diff
@@ -130,11 +137,21 @@ contract MarketRegistry is IMarketRegistry, ClearingHouseCallee, MarketRegistryS
         emit MarketMaxPriceSpreadRatioChanged(baseToken, ratio);
     }
 
-    function setFeeDiscountRatio(address trader, uint24 discountRatio) external checkRatio(discountRatio) onlyOwner {
+    function setFeeDiscountRatio(address trader, uint24 discountRatio)
+        external
+        override
+        checkRatio(discountRatio)
+        onlyFeeManager
+    {
         _feeDiscountRatioMap[trader] = discountRatio;
         emit FeeDiscountRatioChanged(trader, discountRatio);
     }
 
+    function setFeeManager(address accountArg, bool isFeeManagerArg) external onlyOwner {
+        _feeManagerMap[accountArg] = isFeeManagerArg;
+        emit FeeManagerChanged(accountArg, isFeeManagerArg);
```

Comment:
minor feedback: now if we setFeeManager(alice, true) twice, it will emit FeeManagerChanged twice but actually FeeManager is not changed from the 2nd tx, suggest to do any of the following
1. revert if isFeeManagerArg == feeManagerMap[account]
2. only emit when it's diff
3. rename event to sth like....FeeManagerSet?

prefer 1, but if we think owner may want to set many fee manager at once in a batched tx then maybe 2 is easier (1 will break the batched tx easier if owner didn't check the state one by one)

Reply:
(2) seems to be better in your case because it is less intrusive (no-op > revert)